### PR TITLE
feat: add visitors to api-journeys

### PR DESCRIPTION
**Pull request by tataihono**
# Description
Please include a summary of the change and which todo is completed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- [Create visitors connection query](https://3.basecamp.com/3105655/buckets/29689022/todos/5426553649)
- should be able to request a list of visitors in order of most recently created and paginate through them
- visitor record should have the following fields
	- lastChatStartedAt
	- Location
	- User Agent
	- Status
	- Notes
	- Name
	- Email
	- Messenger Network
	- Messenger Identifier

# How should this PR be QA Tested?
## Preconditions
You must get an Authorization Token for the API you are trying to call. Watch this [loom video](https://www.loom.com/share/be774ff9b6b5427089c5a85762161289) for instructions on how to retrieve an Authorization Token. You'll then need to get a GraphQL Client like [Altair](https://altairgraphql.dev/) which you can then run graphql calls on.

## Test Checklist
- [x] call the GraphQL API to retrieve all the visitors via the visitorsConnection query (use teamId 'jfp-team')
	- [example basic query](https://gist.github.com/tataihono/f78af628d2257ccc4d1214e142601d39)
- [x] call the GraphQL API to retrieve visitors limited to 10 results (use the first parameter on the visitorsConnection)
- [x] call the GraphQL API to retrieve visitors on the second page (use the after parameter on the visitorsConnection)

# Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged into main

### CODE REVIEW

File: apps/api-journeys/src/app/modules/journey/journey.resolver.ts [hunk 1]
```diff
@@ -177,18 +179,29 @@ export class JourneyResolver {
...
           status: JourneyStatus.draft,
+          teamId: team.id,
           ...input
         })
         await this.userJourneyService.save({
           userId,
           journeyId: journey.id,
           role: UserJourneyRole.owner
         })
+        await this.memberService.save(
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/journey/journey.resolver.ts:197]: we should add the creating user to the team if they are not already in the team.

File: apps/api-journeys/src/app/modules/userJourney/userJourney.resolver.ts [hunk 2]
```diff
@@ -83,6 +85,22 @@ export class UserJourneyResolver {
...
     const userJourney: UserJourney = await this.getUserJourney(id)
     await this.checkOwnership(userJourney.journeyId, userId)
+
+    const journey = await this.journeyService.get<{ teamId: string }>(
+      userJourney.journeyId
+    )
+
+    console.log()
+
+    await this.memberService.save(
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/userJourney/userJourney.resolver.ts:93]: if we approve a user to manage the journey we should add them to the team.

File: apps/api-journeys/src/app/modules/visitor/visitor.graphql [hunk 3]
```diff
@@ -0,0 +1,75 @@
+"""
+A visitor with attributes connected to a team.
+"""
+type Visitor @key(fields: "id") {
+  id: ID!
+  teamId: ID!
+  userId: ID!
+  createdAt: String!
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/visitor/visitor.graphql:8]: we will need to come back here to add in more fields

File: apps/api-journeys/src/app/modules/visitor/visitor.resolver.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  ) {}
+
+  @Query()
+  async visitorsConnection(
+    @CurrentUserId() userId: string,
+    @Args('teamId') teamId: string,
+    @Args('first') first?: number | null,
+    @Args('after') after?: string | null
+  ): Promise<VisitorsConnection> {
+    const memberResult = await this.memberService.getMemberByTeamId(
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/visitor/visitor.resolver.ts:22]: perhaps this logic will need to eventually become a decorator function like user roles is now.

File: apps/api-journeys/src/app/modules/visitor/visitor.graphql [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  Information to aid in pagination.
+  """
+  pageInfo: PageInfo!
+}
+
+extend type Query {
+  """
+  A list of visitors that are connected with a specific team.
+  """
+  visitorsConnection(
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/visitor/visitor.graphql:61]: this is not a true implementation of a relay connection. Missing before, hasPreviousPage etc.

**Review by csiyang — APPROVED** [review]

**Pull request merged**